client/src/main/java/com/battleships/start_window/connection/Connection.java

```diff
@@ -1,60 +1,93 @@
 package com.battleships.start_window.connection;
 
+import com.battleships.Commands;
+import com.battleships.LogMessages;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.net.Socket;
+import java.util.Optional;
 import java.util.Scanner;
 
 public enum Connection {
     INSTANCE;
-
-    private Socket socket;
+    private Optional<Socket> socket = Optional.empty();
     private PrintWriter socketWriter;
     private Scanner socketScanner;
     private final static Logger logger = LogManager.getLogger(Connection.class);
+    private Thread readCommandsFromUserThread;
 
     public void disconnect() {
         if (isConnected()) {
-            try {
-                sendToServer("quit");
-                socket.close();
-                logger.error("Rozłączony z serwerem na żądanie gracza");
-            } catch (IOException e) {
-               logger.error("Problem z rozłączeniem serwera: " + e.getMessage());
-            } finally {
-                socket = null;
-            }
+            tryToDisconnectFromServer();
+        }
+    }
+
+    private void tryToDisconnectFromServer() {
+        try {
+            sendToServer(Commands.STOP_PLAYING.name());
+            disconnectPlayerOrLogIfFailed();
+        } catch (IOException e) {
+            logger.error(LogMessages.PROBLEM_WHEN_TRYING_TO_DISCONNECT + e.getMessage());
+        } finally {
+            socket = Optional.empty();
+        }
+    }
+
+    private void disconnectPlayerOrLogIfFailed() throws IOException {
```

Comment:
I don't know if Log in function name is needed

Reply:
There is no "Log in" function